addons/escoria-core/game/core-scripts/esc/esc_object_manager.gd

```diff
@@ -439,8 +447,7 @@ func _is_current_room(container: ESCRoomObjects) -> bool:
 # **Returns** True iff container represents the the object manager entry specified
 # by room_key.
 func _compare_container_to_key(container: ESCRoomObjects, room_key: ESCRoomObjectsKey) -> bool:
-	return container.room_global_id == room_key.room_global_id \
-		and container.room_instance_id == room_key.room_instance_id
+	return container.room_global_id == room_key.room_global_id
```

Comment:
I'm curious about this change. I remember making this method to check this very specific combination in case we ever have two rooms setup but that are different instances. What benefit do we gain by removing that part of the check? I don't think it does any harm to leave it as it was.

Reply:
This is because when I save the ESCItem state in the Dictionary, the key is composed of `global_id` and `room_instance_id`. However, when I exit the room to enter another one, then go back to the first room, the `room_instance_id` saved as key is different from the new `room_instance_id` of the room I just went back.

I can revert this change, but then I'll have to create a new test function that doesn't test the instance id.

Reply:
In other words, since the room_object container is no longer deleted when you leave a room if an object's state has changed, room equality is now an issue. Got it. Will take a look.